Content.Client/Store/Ui/StoreMenu.xaml.cs

```diff
@@ -68,22 +72,32 @@ public void UpdateBalance(Dictionary<ProtoId<CurrencyPrototype>, FixedPoint2> ba
         WithdrawButton.Disabled = disabled;
     }
 
-    public void UpdateListing(List<ListingData> listings)
+    public void UpdateListing(List<ListingData> listings, StoreDiscountData[] discounts)
     {
         _cachedListings = listings;
+        _cachedDiscounts = discounts;
```

Comment:
Just put the discounts + string updates into the listing itself rather than adding a heap of complexity to this UI.

Reply:
ListingData is base type for prototype, which i feel complicates things? Will ask in discord a bit later about that, i think...